### PR TITLE
RATEST-329: Update clinical visit workflow test

### PR DESCRIPTION
**Pull request by Irenyak1**
Ticket ID -> [RATEST-329](https://issues.openmrs.org/browse/RATEST-329)

Description -> Updated clinical visit workflow test with the functionality of the Clear Forms button when attaching the patient's support file.

### CODE REVIEW

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature [hunk 1]
```diff
@@ -47,6 +47,12 @@ Feature: Clinical Visit Management
     And a user attaches patient supporting file
     And a user clicks the upload file button
     Then the system uploads the file
+    # User story: Use of clear forms button while attaching document
+    When a user clicks on Attachments link from patient visits dashboard
+    Then the system loads Attachments page
```
Comment (sherrif10) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature:52]: You mean these workflows were missing ?

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature:52]: All firefox is failing.

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature [hunk 2]
```diff
@@ -47,6 +47,12 @@ Feature: Clinical Visit Management
...
+    And a user attaches patient supporting file
```
Comment (mherman22) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature:53]: what happens when the user attaches a file?

Comment (Irenyak1) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature:53]: > what happens when the user attaches a file?

@mherman22 did you have a look at the next step on line 54?

Comment (mherman22) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature:53]: confusion is on me, i confused `clear forms` for an action that happens after clicking the upload button.

Comment (Irenyak1) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature:53]: > confusion is on me, i confused `clear forms` for an action that happens after clicking the upload button.

Have a cup of some hot ginger and lemon. :-)

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature (hunk 1 above)
Comment (Irenyak1) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature:52]: It is pointing to[ add user roles ](https://github.com/openmrs/openmrs-contrib-qaframework/runs/6926994434?check_suite_focus=true#step:7:15840). There's an element that is causing the instability.

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature (hunk 2 above)
Comment (Irenyak1) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature:53]: @mherman22 , @sherrif10 , @kdaud  are there any more reviews on this issue? Let me make a commit to trigger a build and see if the firefox tests will pass.

Comment (mherman22) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature:53]: none from me
cc: @sherrif10 @kdaud

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature:53]: @Irenyak1  Fix the visibility of elements that are breaking the visit note feature file.

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature [hunk 3]
```diff
@@ -47,6 +47,12 @@ Feature: Clinical Visit Management
...
+    And a user clicks the clear forms button
```
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature:54]:
```suggestion
    And a user clicks on clear forms button
```

**Comment by kdaud** [comment]
Could you update this commit with changes from master?

**Comment by Irenyak1** [comment]
> Could you update this commit with changes from master?

Yes @kdaud  I have attended to the reviews. You may make more. Thank you.

**Review by kdaud — APPROVED** [review]

**Comment by Irenyak1** [comment]
Hello @kdaud , @sherrif10 , is there any reason why this can not get into master?

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java [hunk 4]
```diff
@@ -250,6 +250,18 @@ public void systemAddsSupportingFile() {
 		visitsDashboardPage = (PatientVisitsDashboardPage) dashboardPage.goToRecentVisits();
 	}
 
+	@And("a user clicks on clear forms button")
```
Comment (sherrif10) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java:253]: Thanks @Irenyak1  Looks good

**Review by sherrif10 — APPROVED** [review]

**Pull request merged**